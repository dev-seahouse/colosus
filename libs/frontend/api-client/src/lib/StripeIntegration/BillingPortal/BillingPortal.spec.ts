import StripeIntegrationBillingPortalApi from './BillingPortal';

describe('StripeIntegrationBillingPortalApi', () => {
  const stripeIntegrationBillingPortalApi =
    new StripeIntegrationBillingPortalApi();

  describe('createBillingPortalSession()', () => {
    it('should return a valid response', async () => {
      const res =
        await stripeIntegrationBillingPortalApi.createBillingPortalSession();

      expect(res.data).toMatchSnapshot();
    });
  });
});
