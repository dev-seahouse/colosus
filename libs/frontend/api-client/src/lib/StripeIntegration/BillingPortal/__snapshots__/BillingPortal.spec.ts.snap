// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`StripeIntegrationBillingPortalApi > createBillingPortalSession() > should return a valid response 1`] = `
{
  "configuration": "bpc_1MqdTLFabpOjRn7HkkDptkDE",
  "created": 1681801980,
  "customer": "cus_NjaxQisg68yfMP",
  "flow": null,
  "id": "bps_1My8imFabpOjRn7HotpzaMsH",
  "livemode": false,
  "locale": null,
  "object": "billing_portal.session",
  "on_behalf_of": null,
  "return_url": "http://localhost:9000/",
  "url": "https://billing.stripe.com/p/session/test_YWNjdF8xTW9NZ3RGYWJwT2pSbjdILF9OamJ3a3Z2NDdoTDdBSDNJWjdIRUJ0ZHpmZ3FIZ3Vr0100q4zg56E1",
}
`;
